pandas/core/strings.py

```diff
@@ -323,15 +328,24 @@ def str_replace(arr, pat, repl, n=-1, case=True, flags=0):
 
     n : int, default -1 (all)
         Number of replacements to make from start
-    case : boolean, default True
-        If True, case sensitive
+    case : boolean, default None
+        - If True, case sensitive
+        - Defaults to True if `pat` is a string
+        - Must be None if `pat` is a compiled regex
```

Comment:
This sounds like you have to set it to None yourself if passing a compiled regex, while you actually just shouldn't use it. So I would make this something like "Can not be set if `pat` is a compiled regex."